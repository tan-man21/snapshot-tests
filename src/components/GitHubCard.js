import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import React, {useEffect, useState} from 'react';

function GitHubCard() {
    const [gitHubName, setGitHubName] = useState('')
    const [gitHubImg, setGitHubImg] = useState('')

    useEffect(() => {
        fetch('https://api.github.com/users/tan-man21')
        .then(res => res.json())
        .then(data => {
            setGitHubName(data.name)
            setGitHubImg(data.avatar_url)
        })
    }, [])

    return (
        <Card style={{width: '18rem'}}>
            <Card.Img variant='top' src={gitHubImg} />
            <Card.Body>
                <Card.Title>{gitHubName}</Card.Title>
                <Card.Text>
                    This should be the bio!
                </Card.Text>
            </Card.Body>
        </Card>
    )
}

export default GitHubCard