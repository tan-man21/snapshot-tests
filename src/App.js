import logo from './logo.svg';
import './App.css';
import GitHubCard from './components/GitHubCard';

function App() {
  return (
    <div className="App">
        <img src={logo} className="App-logo" alt="logo" />
        <GitHubCard />
    </div>
  );
}

export default App;
