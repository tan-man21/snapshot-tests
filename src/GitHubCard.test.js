import GitHubCard from './components/GitHubCard';
import renderer from 'react-test-renderer';
import App from './App'
import { render, screen, waitFor } from '@testing-library/react';

test('receives GitHub info for GitHubCard', async () => {
    fetch.mockResponseOnce(JSON.stringify({name: 'Tanner Rogers', avatar_url: 'https://avatars.githubusercontent.com/u/147564784?v=4'}))
    render(<App />)
    const gitHubName = await waitFor(() => screen.getByRole('heading', {level: 2}))
    expect(gitHubName).toHaveTextContent('Tanner Rogers')
})

beforeEach(() => {
    // sets everything back to initial state before each test
    fetch.resetMocks();
  })
  
  test('renders a snapshot of GitHubCard', () => {
    const tree =renderer.create(<GitHubCard />).toJSON()
    expect(tree).toMatchSnapshot()
  })