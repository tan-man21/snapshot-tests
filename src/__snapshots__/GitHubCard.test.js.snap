// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot of GitHubCard 1`] = `
<div
  className="card"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <img
    className="card-img-top"
    src=""
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    />
    <p
      className="card-text"
    >
      This should be the bio!
    </p>
  </div>
</div>
`;
